src/JustSaying/AwsTools/MessageHandling/SqsQueueBase.cs

```diff
@@ -165,11 +179,75 @@ private static ServerSideEncryption ExtractServerSideEncryptionFromQueueAttribut
             {
                 return null;
             }
+
             return new ServerSideEncryption
             {
                 KmsMasterKeyId = queueAttributes[JustSayingConstants.AttributeEncryptionKeyId],
                 KmsDataKeyReusePeriodSeconds = queueAttributes[JustSayingConstants.AttributeEncryptionKeyReusePeriodSecondId]
             };
         }
+
+        public async Task<IList<Message>> GetMessagesAsync(
+            int maximumCount,
+            IEnumerable<string> requestMessageAttributeNames,
+            CancellationToken stoppingToken)
+        {
+            var request = new ReceiveMessageRequest
+            {
+                QueueUrl = Uri.AbsoluteUri,
+                MaxNumberOfMessages = maximumCount,
+                WaitTimeSeconds = 20,
```

Comment:
This is an SQS thing, we say we want to poll the receive endpoint for the maximum length of time (20 seconds). This value is validated by the sdk and the sqs endpoint.
The sqs endpoint will try to stop the request at 20 seconds if `MaxNumberOfMessages` isn't reached, but in reality it can spike above that, I've seen 40 seconds before. And in previous versions of the AWSSDK we've seen the tasks never complete (hence the 5min cancallationtoken), this has been solved now but we've kept the defensive code for good measure.

Reply:
Thanks for the explanation. Perhaps we'd still want this to be configurable though?